src/runtime/threads/thread.cpp

```diff
@@ -36,17 +34,11 @@ namespace hpx
 {
     void thread::terminate(const char* function, const char* reason) const
     {
-        try {
-            // free all registered exit-callback functions
-            threads::free_thread_exit_callbacks(id_);
+        // free all registered exit-callback functions
+        threads::free_thread_exit_callbacks(id_);
 
-            // report the error globally
-            HPX_THROW_EXCEPTION(invalid_status, function, reason);
-        }
-        catch(...) {
-            hpx::report_error(std::current_exception());
-            /* nothing else we can do */;
-        }
```

Comment:
Great, thanks! I'll check.

Reply:
I can't see why we would need this anymore. I've added a test which throws from `thread` (`join` a thread twice) which required releasing some locks but otherwise works well.